src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,164 @@
+     * @param input Input to be parsed.
+     * @param isLoading Boolean to indicate if the Duke bot is loading.
+     */
+    public void parse(Duke d, String input, boolean isLoading) {
+        if (input.equals(this.bye)) {
+            System.out.println("Byebye! See you again soon!");
+            d.stop();
+        } else if (input.equals(this.list)) {
+            d.getList();
+        } else if (input.startsWith(this.mark + this.space)) {
+            try {
+                String parameter = input.substring((this.mark + this.space).length());
+                int param = Integer.parseInt(parameter);
+                d.markTask(param);
+            } catch (NumberFormatException e) {
+                System.err.println("Invalid task.");
+            }
+        } else if (input.startsWith(this.unmark + this.space)) {
+            try {
+                String parameter = input.substring((this.unmark + this.space).length());
+                int param = Integer.parseInt(parameter);
+                d.unmarkTask(param);
+            } catch (NumberFormatException e) {
+                System.err.println("Invalid task.");
+            }
+        } else if (input.startsWith(this.todo + this.space)) {
+            String s = input.substring((this.todo + this.space).length());
+            if (s.isBlank()) {
+                System.out.println("The task is empty, what do you really mean?");
+            } else {
+                d.addTodo(s);
+                try {
+                    d.getStorage().writeToFile(this.todo + this.space + s + "\n", isLoading);
+                } catch (IOException e) {
+                    System.err.println("This command could not be saved due to an unknown error.");
+                }
+            }
+        } else if (input.startsWith(this.deadline + this.space)) {
+            String s = input.substring((this.deadline + this.space).length());
+            if (!s.contains(this.space + this.deadlineBy + this.space)) {
+                if (s.startsWith(this.deadlineBy)) {
+                    System.out.println("The task is empty, what do you really mean?");
+                } else {
+                    System.out.println("The deadline is empty, do you mean it has no deadline?");
+                    System.out.println("If it is, please add it as a todo instead.");
+                }
+            } else {
+                String task = s.substring(0, s.indexOf(this.deadlineBy) - this.space.length());
+                String deadline = s.substring(s.indexOf(this.deadlineBy) + this.deadlineBy.length()
+                        + this.space.length());
+                if (task.isBlank()) {
+                    System.out.println("The task is empty, what do you really mean?");
+                } else if (deadline.isBlank()) {
+                    System.out.println("The deadline is empty, do you mean it has no deadline?");
+                    System.out.println("If it is, please add it as a todo instead.");
```

Comment:
I like the user feedback you give here, I can see how these prompts would be very helpful for a first time user.